slitlessutils/core/modules/extract/single/single.py

```diff
@@ -115,14 +115,13 @@ def __init__(self, extorders, mskorders='all', savecont=False, root=None,
         self.writecsv = writecsv
         if self.writecsv:
             self.csvpath = self.outpath+'CSVFILES'+os.sep
-            try:
-                os.mkdir(self.csvpath)
-            except FileNotFoundError:
-                LOGGER.warning('Cannot make CSVPATH, so no CSV files will be written')
-                self.writecsv = False
-                self.csvpath = None
-
-        print(self.outpath, self.writecsv)
+            if not os.path.exists(self.csvpath):
+                try:
+                    os.mkdir(self.csvpath)
+                except FileNotFoundError:
+                    LOGGER.warning('Cannot make CSVPATH, so no CSV files will be written')
```

Comment:
```suggestion
                    LOGGER.warning(f'Cannot make CSVPATH {self.csvpath},'
                                   ' so no CSV files will be written')
```

May as well give the path that's failing here.

Reply:
totally.  Good idea.